src/features/device-state/index.ts

```diff
@@ -45,6 +46,12 @@ export const setup = (app: Application) => {
 		statePatchV2,
 	);
 	app.patch('/device/v3/state', apiKeyMiddleware, statePatchV3);
+	app.get('/device/v3/fleet-state/:fleetUuid', apiKeyMiddleware, fleetStateV3);
+	app.get(
+		'/device/v3/fleet-state/:fleetUuid/release/:releaseUuid',
```

Comment:
I know there was discussion about how the endpoints should look like and the result was what you have, but seeing them now I think the following is much better:

```suggestion
	app.get('/device/v3/fleet/:fleetUuid/state', apiKeyMiddleware, fleetStateV3);
	app.get(
		'/device/v3/fleet/:fleetUuid/release/:releaseUuid/state',
```

Because:

- it does not imply that the endpoint is about "fleet state" when a specific release is requested (i.e. the second endpoint)
- it follows a little more closely the original format of the state endpoint, where the URL ends with `/state`.

Reply:
adopted by @fisehara (https://balena.zulipchat.com/#narrow/stream/332163-_all/topic/curious.20about.20preloading.20efforts/near/308394690)